docs/architecture-decision-records/2023-03-integration-with-station.md

```diff
@@ -0,0 +1,232 @@
+# Zinnia in Filecoin Station
+
+<!--
+This is a minimal template. Feel free to add more sections as needed.
+
+Please review also the Design Doc template and add any relevant sections to your ADR:
+https://www.notion.so/pl-strflt/Writing-a-Design-Doc-aa6034be43c2434ba88a2fd844516e94
+-->
+
+> Status: ACCEPTED
+
+<!--
+PROPOSED, ACCEPTED, REJECTED, DEPRECATED, SUPERSEDED BY {link-to-ADR}
+-->
+
+## Context
+
+Zinnia is a runtime for Filecoin Station modules. It's time to integrate Zinnia into the Station to
+enable the deployment of the first modules.
+
+Zinnia is designed to run multiple modules inside the same process, using V8 Isolates for keeping
+modules cleanly separated and sandboxed. The integration with libp2p (and later IPFS) is designed to
+allow multiple modules to share the same underlying set of network connections and block storage.
+
+Zinnia has two primary modes of operation:
+
+1. A developer tool for building a single module. We want to optimize for ease of use and a fast
+   feedback loop.
+
+   In this mode, Zinnia is typically started and stopped frequently, running for short periods at a
+   time, executing a single module only.
+
+   As a CLI tool, it reads the configuration from a config file (either user-level or
+   project-specific) or CLI arguments. There should be reasonable defaults tailored to developers
+   building a module - e.g. the state files should be stored in the project's working directory.
+
+2. A deployment platform, running modules inside Filecoin Station instances operated by
+   non-technical users. We want to optimize for a smooth user experience for Station operators
+   first, and easy troubleshooting of operations by module developers second.
+
+   In this mode, Zinnia is a long-running process executing multiple modules. It needs to support
+   reloading individual modules when a new version of a module is deployed. It also needs to report
+   various information back to the Station, e.g. activity log and the number of jobs completed.
+
+   As a service, it should read the configuration from environment variables (see
+   [The Twelve-Factor App](https://12factor.net/config)) and allow embedders to improve security
+   e.g. by keeping modules in a different place from the runtime state.
+
+### `zinniad`
+
+For the reasons above, I propose to build a new binary called `zinniad` that will execute Zinnia
+modules inside the Station.
+
+_(This is a strawman proposal, subject to changes during implementation.)_
+
+```bash
+# Address of Station's built-in wallet
+FIL_WALLET_ADDRESS=f1etc \
+# Where to keep state files
+STATE_ROOT=$HOME/Library/Caches/Filecoin\ Station/zinnia \
+# The program to run. The only argument is path to the directory containing module files.
```

Comment:
What does this mean? Does the module folder have one folder per module, each having one executable per arch? Either the configuration needs to be backed into zinniad, or there is convention that everyone adheres to, or a configuration file format. Wdyt?

Reply:
Updated the proposal - see [71994ec](https://github.com/filecoin-station/zinnia/pull/121/commits/71994ec84e91e380c45005db6330df596aecd45c)